src/MobSpawner.cpp

```diff
@@ -242,46 +238,46 @@ bool cMobSpawner::CanSpawnHere(cChunk * a_Chunk, Vector3i a_RelPos, eMonsterType
 		case mtOcelot:
 		{
 			return (
-				(targetBlock == E_BLOCK_AIR) &&
-				(blockAbove == E_BLOCK_AIR) &&
+				(TargetBlock == E_BLOCK_AIR) &&
+				(BlockAbove == E_BLOCK_AIR) &&
 				(
-					(blockBelow == E_BLOCK_GRASS) || (blockBelow == E_BLOCK_LEAVES) || (blockBelow == E_BLOCK_NEW_LEAVES)
+					(BlockBelow == E_BLOCK_GRASS) || (BlockBelow == E_BLOCK_LEAVES) || (BlockBelow == E_BLOCK_NEW_LEAVES)
 				) &&
 				(a_RelPos.y >= 62) &&
-				(random.RandBool(2.0 / 3.0))
+				(Random.RandBool(2.0 / 3.0))
```

Comment:
here

Reply:
The intention is to make certain mobs more rare, although I'm not sure if these values are taken from some documentation or just tweaked until they felt right.

Reply:
I think this doesn't feel right doing this kind of check in that function.

I would perform this operation in the cMobSpawner::TryToSpawnHere function and/or make this customizeable

Reply:
I agree. I think the whole mob spawning code would benefit from a proper review, as it has been largely unchanged since 2013 and most certainly has differences compared to vanilla nowadays. There are still outdated TODOs [like this](https://github.com/cuberite/cuberite/blob/master/src/Chunk.cpp#L587) in a few places.

Definitely something for a new PR though.

Reply:
So I'm gonna keep it like that for now and open an issue to do a cleanup on mob spawning